### PR TITLE
[runtime] Don't abort .cctors. Fixes #43320

### PR DESCRIPTION
**Pull request by kumpera**
This fixes one of the sources of instability on jenkins.

### CODE REVIEW

**Comment by kumpera** [comment]
Broken PR. DON'T MERGE.

**Comment by kumpera** [comment]
@lambdageek please review

File: mono/metadata/threads.c [hunk 1]
```diff
@@ -4794,6 +4793,13 @@ async_abort_critical (MonoThreadInfo *info, gpointer ud)
 	if (mono_get_eh_callbacks ()->mono_install_handler_block_guard (mono_thread_info_get_suspend_state (info)))
 		return MonoResumeThread;
 
+	/*
+	The target thread is running at least one .cctor, which must not be interrupted, so we give up.
```
Comment (lambdageek) [inline mono/metadata/threads.c]: Stale comment - this is about abort-protected blocks now, not just .cctors

**Comment by lambdageek** [comment]
Don't you also need to respect the abort-block count in `self_abort_internal`, too?

**Comment by kumpera** [comment]
No, self aborts right now ignore the protection mechanism. There's a test for that particular behavior.

It kinda makes sense. You should know what you're doing if you self abort.

**Comment by lambdageek** [comment]
Ok, LGTM

**Pull request merged**